packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -103,6 +104,8 @@ export interface AngularCompilerPluginOptions {
   nameLazyFiles?: boolean;
   logger?: logging.Logger;
   directTemplateLoading?: boolean;
+  discoverLazyRoutes?: boolean;
```

Comment:
Can you kindly explain the reasoning behind this option?

Reply:
When using the `loadChildren` string syntax, `@ngtools/webpack` must query `@angular/compiler-cli` via a private API to know which lazy routes exist. This increases build and rebuild time.

When using Ivy, the string syntax is not supported at all. Thus we shouldn't attempt that. 

This commit also exists in https://github.com/angular/angular-cli/pull/13700 for slightly different reasons. There, it is meant to disable this sort of processing for compilations that do not need it.

Reply:
Thanks for the clarification, it makes more sense now :)

Reply:
Should we warn (or probably error) in the transformer (or somewhere else) if the string form is found when this option is off?

Users may not know why their applications are not working.

Reply:
I think not because that means we need to always check, which incurs a performance penalty on every AOT build.

The "right way" of doing that probably a Ivy migration when it's stable.

Reply:
let's try to generally answer questions like this in the form of a comment in the code - no one later will find this thread

Reply:
Done.